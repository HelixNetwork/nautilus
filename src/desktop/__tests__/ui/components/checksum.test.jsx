import React from 'react';
import { shallow } from 'enzyme';

import Checksum from 'ui/components/checksum';

const props = {
    address: 'a28c44524d557d2276170f1e662a9a3abccebb7cc6559a0c00b628613793a710125dc65c',
};

describe('Checksum component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Checksum {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Address value', () => {
        const wrapper = shallow(<Checksum {...props} />);

        expect(wrapper.text()).toEqual(
            'a28c44524d557d2276170f1e662a9a3abccebb7cc6559a0c00b628613793a710125dc65c',
        );
    });

    test('Checksum highlight value', () => {
        const wrapper = shallow(<Checksum {...props} />);
        expect(wrapper.find('mark').text()).toEqual('125dc65c');
    });
});
