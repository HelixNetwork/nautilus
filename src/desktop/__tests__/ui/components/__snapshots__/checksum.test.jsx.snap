// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Checksum component Render the Checksum component 1`] = `
<Fragment>
  a28c44524d557d2276170f1e662a9a3abccebb7cc6559a0c00b628613793a710
  <mark>
    125dc65c
  </mark>
</Fragment>
`;
